compiler/parser/parser.cpp

```diff
@@ -674,19 +679,33 @@ static bool containsOnlyModules(BlockStmt* block, const char* path) {
     }
   }
 
-  if (hasUses == true && hasOther == false && moduleDefs == 1) {
+  if ((hasUses == true || hasRequires == true) &&
+      hasOther == false &&
+      moduleDefs == 1) {
```

Comment:
I think this would be easier to read as nested conditionals, of pulling some parts of the conditional out as variables. But that's a style issue & it's OK this way.